src/classes/ERR_Handler_API.cls

```diff
@@ -88,9 +88,27 @@ global class ERR_Handler_API {
     * @description Creates an error record from an exception, stores it in the database (if enabled), and sends error
     *              notifications (if enabled).
     * @param e Exception to create error record from.
-    * @return context The context the exception was raised on.
+    * @param context The context the exception was raised on.
     */
     global static void processError(Exception e, String context) {
         ERR_Handler.processError(e, context);
     }
+
+    /******************************************************************************************************
+     * @description Creates error records from DML results, stores in database (if enabled) and sends error
+     *              notification (if enabled)
+     * @param dmlResult The results of a single DML operation performed using the Database class.
+     * @param records The records the DML operation was performed on.
+     * @param context The context the errors were raised on
+     */
+
+    global static void processDMLError(List<Object> dmlResult, List<SObject> records, String context) {
```

Comment:
Since this accepts a collection of SObjects, I recommend a method name of `processDMLErrors()`.